grammar.js

```diff
@@ -1,261 +1,170 @@
-        optional(alias(/[^']+/, $.attribute_value)),
-        "'"
+    quoted_attribute_value: ($) =>
+      choice(
+        seq("'", optional(alias(/[^']+/, $.attribute_value)), "'"),
+        seq('"', optional(alias(/[^"]+/, $.attribute_value)), '"')
       ),
+
+    directive: ($) =>
       seq(
-        '"',
-        optional(alias(/[^"]+/, $.attribute_value)),
-        '"'
-      )
-    ),
-
-    directive: $ => seq(
-      choice('<%', '<%=', '<%%', '<%%='),
-      prec.left(
-        seq(
-          choice($.partial_expression_value, $.expression_value),
-          '%>',
+        choice("<%", "<%=", "<%%", "<%%="),
+        prec.left(
+          seq(choice($.partial_expression_value, $.expression_value), "%>")
         )
-      )
-    ),
+      ),
 
-    comment: $ => choice($._html_comment, $._bang_comment, $._hash_comment),
+    comment: ($) => choice($._html_comment, $._bang_comment, $._hash_comment),
 
-    _html_comment: $ => seq(
-      '<!--',
-      prec.left(
-        seq(
-          repeat(/[^-]+|-/),
-          '-->'
-        )
-      )
-    ),
+    _html_comment: ($) => seq("<!--", prec.left(seq(repeat(/[^-]+|-/), "-->"))),
 
-    _bang_comment: $ => seq(
-      '<%!--',
-      prec.left(
-        seq(
-          repeat(/[^-]+|-/),
-          '--%>'
-        )
-      )
-    ),
+    _bang_comment: ($) =>
+      seq("<%!--", prec.left(seq(repeat(/[^-]+|-/), "--%>"))),
 
-    _hash_comment: $ => seq(
-      '<%#',
-      prec.left(
-        seq(
-          repeat($._code),
-          '%>'
-        )
-      )
-    ),
+    _hash_comment: ($) => seq("<%#", prec.left(seq(repeat($._code), "%>"))),
 
-    expression_value: $ => repeat1($._code),
+    expression_value: ($) => repeat1($._code),
 
-    partial_expression_value: $ => seq(
-      choice(
-        seq(/end[\)\]\}]*/, repeat($._code)),
-        seq(repeat($._code), choice('do', '->'), optional(seq('#', repeat($._code)))),
+    partial_expression_value: ($) =>
+      seq(
+        choice(
+          seq(/end[\)\]\}]*/, repeat($._code)),
+          seq(
+            repeat($._code),
+            choice("do", "->"),
+            optional(seq("#", repeat($._code)))
+          )
+        )
       ),
-    ),
 
-    component_name: $ => choice(
-      seq(optional($.module), seq(".", $.function)),
-      $.module
-    ),
+    component_name: ($) =>
+      choice(seq(optional($.module), seq(".", $.function)), $.module),
 
-    module: $ => /([A-Z][^\-<>{}!"'/=\s\.]*)(\.[A-Z][^\-<>{}!"'/=\s\.]*)*/,
+    module: ($) => /([A-Z][^\-<>{}!"'/=\s\.]*)(\.[A-Z][^\-<>{}!"'/=\s\.]*)*/,
 
-    function: $ => /[a-z][^\-<>{}!"'/=\s\.]*/,
+    function: ($) => /[a-z][^\-<>{}!"'/=\s\.]*/,
 
-    _code: $ => /[^%\s]+|[%\s]/,
+    _code: ($) => /[^%\s]+|[%\s]/,
 
-    tag_name: $ => /[a-z]+[^<>{}!"'/=\s]*/,
+    tag_name: ($) => /[a-z]+[^<>{}!"'/=\s]*/,
 
-    attribute_name: $ => token(prec(-1, /[^<>{}"'/=\s]+/)),
+    attribute_name: ($) => token(prec(-1, /[^:<>{}"'/=\s][^<>{}"'/=\s]+/)),
```

Comment:
```suggestion
    attribute_name: ($) => token(prec(-1, /[^:<>{}"'/=\s][^<>{}"'/=\s]*/)),
```

Small nit, I think with the `+` you couldn't have a single-character attribute name. I'm not sure single-character attributes actually exist in the wild though 🙃